spec/models/service_spec.rb

```diff
@@ -142,14 +142,77 @@
       let(:other_beginning) { (service_range.begin - 2.months).at_beginning_of_week }
       let(:other_ending) { (service_range.begin - 1.month).at_end_of_week - 2.days }
 
-      before { create :service, civil_servant: civil_servant, beginning: other_beginning, ending: other_ending }
+      context 'when existing service it is a service agreement' do
+        before do
+          create :service, :civil_servant_agreement_pending,
+                 civil_servant: civil_servant, beginning: other_beginning, ending: other_ending
+        end
+
+        context 'when there is no overlapping service' do
+          it { is_expected.to be false }
+        end
+
+        context 'when there is an overlapping service' do
+          let(:other_ending) { service_range.begin.at_end_of_week - 2.days }
+
+          it { is_expected.to be false }
+        end
+      end
+
+      context 'when existing service is a definitive service' do
+        before { create :service, civil_servant: civil_servant, beginning: other_beginning, ending: other_ending }
+
+        context 'when there is no overlapping service' do
+          it { is_expected.to be false }
+        end
+
+        context 'when there is an overlapping service' do
+          let(:other_ending) { service_range.begin.at_end_of_week - 2.days }
+
+          it { is_expected.to be true }
+        end
+      end
+    end
+
+    describe '#legitimate_civil_servant_decision' do
+      context 'when the civil servant has not yet decided' do
+        subject do
+          service = build(:service, :civil_servant_agreement_pending)
+          service.update(civil_servant_agreed: true)
+          service.errors.added?(:civil_servant_agreed, :already_decided)
+        end
```

Comment:
```suggestion
        let(:service) { build(:service, :civil_servant_agreement_pending, civil_servant_agreed: true)

        subject do
          service.tap(&:validate).errors.added?(:civil_servant_agreed, :already_decided)
        end
```

Reply:
Ur suggesten forgoes the needed update

Reply:
Doesn't the trigger also trigger on create?